File: 02. C# Part2/01. Arrays-Homework/07. SelectionSort/SelectionSort.cs
using System;
using System.Collections.Generic;

class SelectionSort
{
    static void Main()
    {
                //Sorting an array means to arrange its elements in increasing order. Write a program to sort an array.
                //Use the Selection sort algorithm: Find the smallest element, move it at the first position, find the
                //smallest from the rest, move it at the second position, etc.

        Console.Write("Enter the lenght of the array: ");
        int n = int.Parse(Console.ReadLine());
        int[] array = new int[n];
        for (int i = 0; i < n; i++)
        {
            array[i] = int.Parse(Console.ReadLine());
        }
        int minValue = int.MaxValue;
        int minIndex = -1;
        for (int i = 0; i < array.Length; i++)
        {
            minValue = int.MaxValue;
        
        for (int j = i + 1; j < array.Length; j++)
        {
            if (array[j] < minValue)
            {
                minValue = array[j];
                minIndex = j;
            }
        }
        int temp = array[i];
        array[i] = minValue;
        array[minIndex] = temp;
        }
        Console.WriteLine(string.Join(", ", array));
    }
}

File: 06. JavaScript Basics/05. Arrays-Homework/05. Arrays-Homework/05. SelectionSort/SelectionSort.js
//Sorting an array means to arrange its elements in increasing order.
//Write a script to sort an array.
//Use the selection sort algorithm: Find the smallest element, move it at the first position, 
//find the smallest from the rest, move it at the second position, etc.

function selectionSort(arr) {
    var i, j, tmp, tmp2;
    for (i = 0; i < arr.length - 1; i++) {
        tmp = i;
        for (j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[tmp]) {
                tmp = j;
            }
        }
        if (tmp != i) {
            tmp2 = arr[tmp];
            arr[tmp] = arr[i];
            arr[i] = tmp2;
        }
    }
    arr = arr.join(", ")
    jsConsole.writeLine("Sorted array: " + arr);
}
selectionSort([4, 3, 6, 8, 9, 2, 11, 0, -4, -2, 18]);